plugins/flytekit-vaex/flytekitplugins/vaex/sd_transformers.py

```diff
@@ -0,0 +1,73 @@
+import os
+import typing
+
+import pandas as pd
+import vaex
+
+from flytekit import FlyteContext, StructuredDatasetType
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.types.structured.structured_dataset import (
+    PARQUET,
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetTransformerEngine,
+)
+
+
+class VaexDataFrameToParquetEncodingHandler(StructuredDatasetEncoder):
+    def __init__(self):
+        super().__init__(vaex.dataframe.DataFrameLocal, None, PARQUET)
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        df = typing.cast(vaex.dataframe.DataFrameLocal, structured_dataset.dataframe)
+        path = ctx.file_access.get_random_remote_directory()
+        local_dir = ctx.file_access.get_random_local_directory()
+        local_path = os.path.join(local_dir, f"{0:05}")
+        df.export_parquet(local_path)
```

Comment:
Apologies for reviewing it late! As per their docs, HDF5 is the most suitable when the data is huge: https://vaex.readthedocs.io/en/docs/example_io.html#id1. We can go with Parquet, just want to give a heads-up.

Reply:
yup, we can register another handler (`VaexDataFrameToHDF5EncodingHandler`), so people can use `Annotated` to update the default format. we can add it in the separate PR

```python
def t1() -> Annotated[StructuredDataset, HDF5]
```

Reply:
Arrow may also be useful to ad support for https://vaex.readthedocs.io/en/latest/faq.html, what are your thoughts ?
Im happy to add support for it in separate PR if thats ok ?

Reply:
That's awesome, works for me! Please feel free to create issues accordingly and let me know! :)